src/mock/service.rs

```diff
@@ -1,29 +1,48 @@
 use std::sync::{Arc, RwLock};
 use std::time::Instant;
 
+use lazy_static::lazy_static;
+use rayon::{ThreadPool, ThreadPoolBuilder};
+
 use log::info;
 
 use anyhow::{anyhow, Context, Error};
 
 use super::compilation::compile_configuration;
 use super::model::{Expectation, IncomingRequest};
+use tokio::sync::oneshot;
 
 pub struct State {
     pub expectations: Vec<Expectation>,
 }
 
 pub type SharedState = Arc<RwLock<State>>;
 
-// TODO add unit tests
-pub fn add_configuration(
+lazy_static! {
+    static ref POOL: ThreadPool = ThreadPoolBuilder::new()
+        .num_threads(3)
+        .stack_size(8 * 1024 * 1024)
+        .build()
+        .unwrap();
+}
+
+// Todo add Unit tests
+pub async fn load_configuration(
     state: SharedState,
     id: String,
     configuration: String,
 ) -> Result<(), Error> {
```

Comment:
to avoid lock issue maybe we could consider to return configuration result and update state externally, at least for configurations provided from files ?

Reply:
I splited this behavior in two separate functions `load_dhall_expectation` to compile dhall content and return an aync result with expectation and `add_expectations_in_state` to add a vec of expectation in state